src/production/models/classifier.py

```diff
@@ -1,55 +1,66 @@
-from .convnext.utils import get_total_class
 from torch import nn
 from PIL import Image
-import streamlit as st
 import matplotlib.pyplot as plt
 import torch
 import numpy as np
 
 
 
-def load_classifier_model()->Tuple[AutoFeatureExtractor, AutoModelForImageClassification]:
+def load_classifier_model(engine: str)->Tuple:
     # load convnext from huggingface hub
-    extractor = AutoFeatureExtractor.from_pretrained("facebook/convnext-base-224-22k")
-    convnext = AutoModelForImageClassification.from_pretrained("facebook/convnext-base-224-22k")
+    if engine=='convnext_v1':
+        extractor = AutoFeatureExtractor.from_pretrained("facebook/convnext-base-224-22k")
+        convnext = AutoModelForImageClassification.from_pretrained("facebook/convnext-base-224-22k")
+    elif engine=='convnext_v2':
+        # this is to test the model published in huggingface hub
+        extractor = AutoImageProcessor.from_pretrained("Pavarissy/ConvNextV2-large-DogBreed")
+        convnext = ConvNextV2ForImageClassification.from_pretrained("Pavarissy/ConvNextV2-large-DogBreed")
+
     return (extractor, convnext)
 
-def init_model(weight_path: str)->nn.Module:
+def init_model(engine:str='convnext_v2', weight_path: str=None)->nn.Module:
     # initilialize a model
-    extractor , convnext = load_classifier_model()
+    if engine=='convnext_v1':
+        extractor , convnext = load_classifier_model(engine)
 
-    conv_block = convnext.convnext
-    predictor = ConvNextTransfer(conv_block)
+        conv_block = convnext.convnext
+        predictor = ConvNextTransfer(conv_block)
 
-    if weight_path:
-        # Load the model's state dictionary
-        state_dict = torch.load(weight_path)
+        if weight_path:
+            # Load the model's state dictionary
+            state_dict = torch.load(weight_path)
 
-        # Load the state dictionary to the model
-        predictor.load_state_dict(state_dict)
-        return predictor, extractor
+            # Load the state dictionary to the model
+            predictor.load_state_dict(state_dict)
+            return predictor, extractor
 
-    else:
-        # in case that there has no weight to be loaded
-        return predictor, extractor
+    elif engine=='convnext_v2':
+        # in case that it is model available on huggingface
+        extractor , convnext = load_classifier_model(engine)
+        return convnext, extractor
     
 def classify(image: Image.Image)->Tuple[torch.Tensor, 
                                         torch.Tensor,
                                         str
                                         ]:
-    # init this by ourselves
-    weight_file = None#'/workspaces/dog-breed-classification-webapp/src/production/models/model_weights/best_model_convnext.pth'
-    _ , idx2label = get_total_class()
 
     # init the model
-    classifier, extractor = init_model(weight_file)
+    classifier, extractor = init_model(engine='convnext_v2')
+    
```

Comment:
we will see that it will **re-init the model everytime** we upload new image, which is time-consuming since the `ConvNext` itself is very large. Consider **optimize the speed** by downloading everything first before making an inference at the end would much help